src/geometry.rs

```diff
@@ -1043,7 +1043,7 @@ pub trait Geom: AsRaw<RawType = GEOSGeometry> {
     ///
     /// assert_eq!(geom.get_num_dimensions(), Ok(2));
     /// ```
-    fn get_num_dimensions(&self) -> GResult<usize>;
+    fn get_num_dimensions(&self) -> GResult<i32>;
```

Comment:
If it returns `-1`, it's supposed to be handled by the binding, not by the user.